src/luarocks/fs/unix/tools.lua

```diff
@@ -57,7 +57,7 @@ end
 -- plus an error message.
 function tools.copy(src, dest, perm)
    assert(src and dest)
-   if fs.execute(vars.CP, src, dest) then
+   if fs.execute(vars.CP.." -PR "..src..dest) then
```

Comment:
```suggestion
   if fs.execute(vars.CP.." -PR ", src, dest) then
```

This PR failed the smoke test. Not sure if the above works portably, but should be worth a try.

Reply:
I think that is it, thank you!